java-openliberty/jpa/README.md

```diff
@@ -0,0 +1,247 @@
+
+Now, the only thing that remains is to connect the DB and the application. We will use odo to create a link to the Dev4Devs PostgreSQL Database Operator in order to access the database connection information.
+
+Display the services available to odo: - You will see an entry for the PostgreSQL Database Operator displayed:
+
+```shell
+> odo catalog list services
+Operators available in the cluster
+NAME                                             CRDs
+postgresql-operator.v0.1.1                       Backup, Database
+>
+```
+use odo to create an odo service for the PostgreSQL Database Operator by entering the previous result in the following format: `<NAME>/<CRDs>`
+```shell
+>  odo service create postgresql-operator.v0.1.1/Database
+```
+push this service instance to the cluster
+```shell
+> odo push
+```
+
+List this service
+```shell
+> odo service list
+NAME                  AGE
+Database/database     6m31s
+
+>
+```
+Create a Service Binding Request between the application and the database using the Service Binding Operator service created in the previous step
+`odo link` command: 
+
+```shell
+> odo link Database/database
+```
+
+push this link to the cluster
+```shell
+> odo push
+```
+
+After the link has been created and pushed a secret will have been created containing the database connection data that the application requires.
+
+You can inspect the new intermediate secret via the Openshift console in Administrator view by navigating to Workloads > Secrets and clicking on the secret named `mysboproj-servicebindingrequest-example-servicebindingrequest` Notice it contains 5 pieces of data all related to the connection information for your PostgreSQL database instance.
+
+Re-deploy the applications using odo
+```shell
+odo push -f
+```
+
+Once the new version is up (there will be a slight delay until application is available), navigate to the CreatePerson.xhtml using the URL created in a previous step. Enter requested data and click the "Save" button
+![Create Person xhtml page](../../assets/createPersonDB.png)
+
+Notice you are re-directed to the PersonList.xhtml page, where your data is displayed having been input to the postgreSQL database and retrieved for display purposes.
+![Create Person xhtml page](../../assets/displayPeople.png)
+
+You may inspect the database instance itself and query the table to see the data in place by using the postgreSQL command line tool, psql.
+
+Navigate to the pod containing your db from the Openshift Console
+
+Click on the terminal tab.
+
+At the terminal prompt access psql for your database
+
+```shell
+sh-4.2$ psql my-demo-db
+psql (12.3)
+Type "help" for help.
+
+my-demo-db=#
+```
+
+Issue the following SQL statement:
+
+```shell
+my-demo-db=# SELECT * FROM testuser.person;
+```
+
+You can see the data that appeared in the results of the test run:
+```shell
+ personid | age |  name   
+----------+-----+---------
+        5 |  52 | person1
+(1 row)
+
+my-demo-db=# 
+```
```

Comment:
Another nice confirmation of your setup is to do this from your odo CLI:

`odo exec -- bash -c 'export | grep DATABASE'`